### PR TITLE
Add support for producer compression

### PR DESCRIPTION
**Pull request by hariso**
### Description

Fixes https://github.com/ConduitIO/conduit/issues/838.

Tested with:
```yaml
---
version: 1.0
pipelines:
  pipeline1:
    status: running
    name: pipeline1
    description: 'Generator source, Kafka destination'
    connectors:
      source1:
        type: source
        plugin: builtin:generator
        name: generator-source
        settings:
          recordCount: "-1"
          readTime: "0s"
          format.options: "/tmp/big-file.txt"
          format.type: "file"
      destination1:
        type: destination
        plugin: "standalone:kafka"
        name: kafka-destination
        settings:
          servers: "localhost:9092"
          topic: "non-existing-topic"
          compression: "gzip"
          batchBytes: "10000000"

```
The test file is 3.4 MB in size.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-kafka/pulls) for the same update/change.
- [ ] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: destination.go [hunk 1]
```diff
@@ -36,60 +36,59 @@ func (d *Destination) Parameters() map[string]sdk.Parameter {
...
 			Description: "SASL username. required if saslMechanism is provided.",
 		},
 		SASLPassword: {
 			Default:     "",
-			Required:    false,
 			Description: "SASL password. required if saslMechanism is provided.",
 		},
+		Compression: {
+			Default:     "",
+			Description: "Compression applied to messages. Possible values: gzip, snappy, lz4, zstd.",
```
Comment (lovromazgon) [inline destination.go:86]: Would be nice if we attach a [`ValidationInclusion`](https://github.com/ConduitIO/conduit-connector-sdk/blob/edae165f49975c04fb5d9e549ea1987060068fc2/validation.go#L131) on this parameter, then the UI displays a dropdown. We should also specify the parameter [type](https://github.com/ConduitIO/conduit-connector-sdk/blob/edae165f49975c04fb5d9e549ea1987060068fc2/specifier.go#L51).

**Review by lovromazgon — APPROVED** [review]
👍

File: destination.go (hunk 1 above)
Comment (hariso) [reply destination.go:86]: Yeah, I was thinking about that too, but I'd like to go over all of the configs and update them all in one go.

Comment (lovromazgon) [reply destination.go:86]: Fine with me 👍

Comment (lovromazgon) [reply destination.go:86]: At that point we could also move to `paramgen` and simplify the parsing of parameters using `sdk.Util.ParseConfig`.

**Pull request merged**